Test_Task/main.rb

```diff
@@ -0,0 +1,96 @@
+require 'watir'
+require 'nokogiri'
+require 'json'
+require 'date'
+require_relative './transactions'
+
+class WebBanking
+  attr_reader :browser, :page, :operation, :desc_operation
+  def initialize
+    @browser = Watir::Browser.new :chrome
+  end
+
+  def parse_accounts
+    sign_in
+    sleep(3)
+    accounts = browser.divs(class: %w(contract status-active))
+    info_list = accounts.map do |div|
+      Watir::Wait.until { div.div(class: "contract-cards").a.present? }
+      div.div(class: "contract-cards").a.click
+      go_to_info
+      go_to_transactions
+      data = {
+          name: parse_name,
+          balance: parse_balance,
+          currency: parse_currency,
+          nature: parse_nature,
+          transactions: get_transaction_info.map(&:to_hash)  #transaction.get_transaction_info(browser)
+      }
+      browser.li(class: %w(new_cards_accounts-menu-item active)).a.click
+      data
+    end
+    JSON.pretty_generate(accounts: info_list)
+  end
+
+  def sign_in
+    browser.goto 'https://wb.micb.md'
+    browser.text_field(class: "username").set("connicov1")
+    browser.text_field(id: "password").set("Paviliong7")
+    browser.button(class: "wb-button").click
+  end
+
+  def go_to_info
+    browser.ul(class: %w(ui-tabs-nav ui-helper-reset ui-helper-clearfix ui-widget-header ui-corner-all)).lis[1].click
+    @page = Nokogiri::HTML.parse(browser.div(id: "contract-information").html)
+  end
+
+  def parse_name #(page)
+    page.css('tr')[6].css(".value").text
+  end
+
+  def parse_balance
+    page.css('tr')[8].css(".value").css('span')[0].text
+  end
+
+  def parse_currency #(page)
+    page.css('tr')[8].css(".value").css('span')[1].text
+  end
+
+  def parse_nature
+    page.css('tr')[3].css(".value").text.gsub("1. ","").gsub("2. ", "")
+  end
+
+  def go_to_transactions
+    browser.ul(class: %w(ui-tabs-nav ui-helper-reset ui-helper-clearfix ui-widget-header ui-corner-all)).lis[2].click
+    Watir::Wait.until { browser.input(class: %w(filter-date from maxDateToday hasDatepicker)).present? }
+    browser.input(class: %w(filter-date from maxDateToday hasDatepicker)).click
+    Watir::Wait.until { browser.a(class: %w(ui-datepicker-prev ui-corner-all)).present? }
+    browser.a(class: %w(ui-datepicker-prev ui-corner-all)).click
+    sleep(3)
+    Watir::Wait.until { browser.a(class: "ui-state-default").present? }
+    browser.a(class: "ui-state-default").click   # I use the first day of the month
+    sleep(3)
+  end
+
+  def get_transaction_info
+    arr = browser.div(class: "operations").lis
+    sleep(3)
+    transaction_list = arr.map do |li|
+      Watir::Wait.until { li.a.present? }
+      li.a.click
+      @operation = Nokogiri::HTML.parse(browser.div(class: "operation-details-body").html)
+      @desc_operation = Nokogiri::HTML.parse(browser.div(class: "operation-details-header").html)
+      browser.send_keys :escape
```

Comment:
very hard to test that.
separate parsing logic and driver logic

add method `parse_transaction` which should return Transaction instance ;)